.eslintrc.cjs

```diff
@@ -1,34 +1,60 @@
+/** @type {import('eslint').Linter.Config} */
 module.exports = {
 	root: true,
 	parser: '@typescript-eslint/parser',
 	extends: [
 		'eslint:recommended',
 		'plugin:@typescript-eslint/recommended',
+		'plugin:svelte/recommended',
 		'plugin:storybook/recommended',
 		'prettier',
 	],
-	plugins: ['svelte3', '@typescript-eslint'],
-	ignorePatterns: ['*.cjs'],
+	plugins: ['@typescript-eslint'],
 	overrides: [
 		{
 			files: ['*.svelte'],
-			processor: 'svelte3/svelte3',
+			parser: 'svelte-eslint-parser',
+			parserOptions: {
+				parser: '@typescript-eslint/parser',
+			},
 		},
 	],
-	settings: {
-		'svelte3/typescript': () => require('typescript'),
-	},
 	parserOptions: {
 		sourceType: 'module',
-		ecmaVersion: 2020,
+		ecmaVersion: 'latest',
+		extraFileExtensions: ['.svelte'],
 	},
 	env: {
 		browser: true,
-		es2017: true,
+		es2024: true,
 		node: true,
 	},
 	rules: {
+		// eslint
 		'no-console': 'warn',
+
+		// @typescript-eslint
+		'@typescript-eslint/no-unused-vars': [
+			'warn',
+			{
+				argsIgnorePattern: '^_',
+				varsIgnorePattern: '^_',
+			},
+		],
+
+		// eslint-plugin-svelte
+		'svelte/no-target-blank': 'error',
+		'svelte/no-immutable-reactive-statements': 'error',
+		'svelte/no-trailing-spaces': 'error',
+		'svelte/prefer-style-directive': 'error',
+		'svelte/no-reactive-literals': 'error',
+		'svelte/no-useless-mustaches': 'error',
+		// TODO: opt in to these at a later stage
+		'svelte/button-has-type': 'off',
+		'svelte/require-each-key': 'off',
+		'svelte/no-at-html-tags': 'off',
+		'svelte/no-unused-svelte-ignore': 'off',
+		'svelte/require-stores-init': 'off',
```

Comment:
These are additional helpful lint rules from eslint-plugin-svelte that I would recommend, but they do produce a lot of new lint issues, so should be adopted incrementally.

https://sveltejs.github.io/eslint-plugin-svelte/rules/button-has-type/ would have been very useful to have enabled from the start to avoid major bug like: https://github.com/melt-ui/melt-ui/pull/442

Reply:
Nice, thanks for the suggestion